x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -792,6 +793,37 @@ export const getSavedObjectTypes = (
         },
       },
     },
+    [OTEL_POLICY_SAVED_OBJECT_TYPE]: {
+      name: OTEL_POLICY_SAVED_OBJECT_TYPE,
+      indexPattern: INGEST_SAVED_OBJECT_INDEX,
+      hidden: false,
+      namespaceType: 'multiple',
+      management: {
+        importableAndExportable: false,
+      },
+      mappings: {
+        properties: {
+          name: { type: 'keyword' },
+          description: { type: 'text' },
+          namespace: { type: 'keyword' },
+          policy_ids: { type: 'keyword' },
+          output_id: { type: 'keyword' },
+          integration: {
+            properties: {
+              name: { type: 'keyword' },
+              version: { type: 'keyword' },
+            },
+          },
+          vars: { type: 'flattened' },
```

Comment:
Interesting, does `flattened` work for complex data types? Or we do some additional processing for these when reading the saved object?

Reply:
Yes it works well for complex data types and it doesn't need additional processing. We're already using it for existing mappings.